src/extra/others/gridnav/lv_gridnav.c

```diff
@@ -0,0 +1,275 @@
+
+
+/**
+ * @file lv_gridnav.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_gridnav.h"
+#include "../../../misc/lv_assert.h"
+#include "../../../misc/lv_math.h"
+#include "../../../core/lv_indev.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef struct {
+    lv_gridnav_ctrl_t ctrl;
+    lv_obj_t * focused_obj;
+} lv_gridnav_dsc_t;
+
+typedef enum {
+    FIND_LEFT,
+    FIND_RIGHT,
+    FIND_TOP,
+    FIND_BOTTOM,
+    FIND_NEXT_ROW_FIRST_ITEM,
+    FIND_PREV_ROW_LAST_ITEM,
+    FIND_FIRST_ROW,
+    FIND_LAST_ROW,
+} find_mode_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void gridnav_event_cb(lv_event_t * e);
+static lv_obj_t * find_chid(lv_obj_t * obj, lv_obj_t * start_child, find_mode_t mode);
+static lv_coord_t get_x_center(lv_obj_t * obj);
+static lv_coord_t get_y_center(lv_obj_t * obj);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_gridnav_add(lv_obj_t * obj, lv_gridnav_ctrl_t ctrl)
+{
+    lv_gridnav_dsc_t * dsc = lv_mem_alloc(sizeof(lv_gridnav_dsc_t));
```

Comment:
I'd suggest calling `lv_gridnav_remove` first as otherwise the behavior will be undefined with two gridnavs on one object.

Reply:
Added.